### PR TITLE
Zephyr sensors: updated to Zephyr v1.6.0 to support unified kernel

### PR DESCRIPTION
**Pull request by kiendinh**
@gvancuts, @ttzeng, @qiuzhong: The Zephyr sensors have been updated to work with Zephyr 1.6.0. Please take a look then.

### CODE REVIEW

**Comment by uhofemei** [comment]
:-1: 

This PR does improve things in that BLE is functional and I can discover the power meter device. There are warning messages about the use of deprecated grove LCD functions and the LCD panel doesn't initialize correctly. 

@kiendinh  regarding the repository of the power meter code in Teamforge, I learned that it is possible to grant OTC members anonymous access to the repository. All that needs to be done is to create a new project ans select the OTC Anonymous Server when creating the project. This will make it a lot easier for the Zephyr engineering team to access and work on TME demo code we keep there.

**Comment by kiendinh** [comment]
@uhofemei The deprecated use is in the lcd driver. It's the problem of Zephyr 1.6 release and has been solved in the Zephyr master branch. However, I found it functioned well here in my setup.

**Comment by qiuzhong** [comment]
@kiendinh, I built the code for this PR on Zephyr 1.6.0. Indeed, there're some warnings as @uhofemei metioned when building the arc part, but I saw the LCD functioning well.

You updated both DC power meter and the environmental sensor.
Here are my test results:

1. The DC power meter worked well.
OCF server power-uart.js could receive the power data via bluetooth from Zephyr 101 board when creating OCF resources.

2. As for the environmental sensor, I flashed another Zephyr 101 board. 
    After updating the bluetooth firmware based on Zephyr 1.6.0, the Zephyr Environmental Sensor can be discovered by a bluetooth client app.

    The OCF server environment_sensor.js can receive the environmental data as well.
    
	The temperature/humidity/pressure values're all normal. However, the UV index value seems invalid again. It ranges from 1x to 2xx and changes dramatically. 
	
	I think it may be the limits of the way to calculate the UV index. Since the UV sensor is in doors, its value might be out of  range of the UV index calculating way. I'll choose a sunny day to check it under the sun. The current weather condition here is not quite suitable for the UV sensor.

File: sensors/DC_power_meter/src/Makefile [hunk 1]
```diff
@@ -2,6 +2,14 @@ ccflags-y += ${PROJECTINCLUDE} \
...
 		-I$(ZEPHYR_BASE)/drivers
 
+ifneq ($(VERSION_MAJOR), 1)
+$(error This app works only with Zephyr 1.6)
+endif
+
+ifneq ($(VERSION_MINOR), 6)
+$(error This app works only with Zephyr 1.6)
+endif
+
```
Comment (gvancuts) [inline sensors/DC_power_meter/src/Makefile]: I have not tested this but in case you use Zephyr 2.1 (hypothetical example since it does not exist yet), would that not spit out the same message twice ("This app works only with Zephyr 1.6")?

Maybe you could nest those (if possible) to avoid this, e.g.:
```
ifneq ($(VERSION_MAJOR), 1)
   ifneq ($(VERSION_MINOR), 6)
   $(error This app works only with Zephyr 1.6)
   endif
endif
```

File: sensors/environmental-sensors/README.md [hunk 2]
```diff
@@ -96,4 +96,4 @@ Please make sure that BLE is enabled on the device running the OCF resource serv
 Use command `hciconfig` to check if it is unblocked and running.
 
 ## Supported Zephyr versions:
```
Comment (gvancuts) [inline sensors/environmental-sensors/README.md]: I'd prefer to have that as the first section, would you mind moving this towards the top?

File: sensors/environmental-sensors/arc/src/Makefile [hunk 3]
```diff
@@ -2,4 +2,12 @@ ccflags-y += ${PROJECTINCLUDE} \
...
 		-I$(ZEPHYR_BASE)/drivers -I$(ZEPHYR_BASE)/arch/arc
 
+ifneq ($(VERSION_MAJOR), 1)
+$(error This app works only with Zephyr 1.6)
+endif
+
+ifneq ($(VERSION_MINOR), 6)
+$(error This app works only with Zephyr 1.6)
+endif
+
```
Comment (gvancuts) [inline sensors/environmental-sensors/arc/src/Makefile]: Same comment about nesting the `ifneq` statements

File: sensors/DC_power_meter/README.md [hunk 4]
```diff
@@ -53,14 +53,14 @@ In order to build the Zephyr app for the Arduino 101 board, first setup the Zeph
...
 ```
 Use the following commands to build and flash the BLE app to the Arduino 101 board:
 ```
-    $ make pristine && make BOARD=arduino_101_factory ARCH=x86
-    $ sudo -E dfu-util -a x86_app -D outdir/zephyr.bin
+    $ make BOARD=arduino_101 ARCH=x86
+    $ sudo -E dfu-util -a x86_app -D outdir/arduino_101/zephyr.bin
 ```
 
 ## Supported Zephyr versions:
```
Comment (gvancuts) [inline sensors/DC_power_meter/README.md]: Could we move this section towards the top, I think it would make it clearer to the user.

File: sensors/environmental-sensors/README.md [hunk 5]
```diff
@@ -56,7 +56,7 @@ Zephyr dev environment following steps:
     $ git clone https://gerrit.zephyrproject.org/r/zephyr && cd zephyr && git checkout tags/[supported_version]
     $ source ./zephyr-env.sh
 ```
-* Flash a bluetooth firmware named [`MyNewt`](https://wiki.zephyrproject.org/index.php?title=Arduino_101&oldid=975#Bluetooth_firmware_for_the_Arduino_101) which is supported by Zephyr
+* Following this [instruction](https://wiki.zephyrproject.org/view/Arduino_101#Bluetooth_firmware_for_the_Arduino_101) to build and flash the Zephyr bluetooth firmware to the board
 
 **Note**: You can learn more information from this [wiki](https://wiki.zephyrproject.org/index.php?title=Arduino_101&oldid=975)
```
Comment (gvancuts) [inline sensors/environmental-sensors/README.md]: This points at a specific version of the wiki page (`oldid=975`), is this still accurate or could we change that to point at the latest current (or better yet, specify the latest current by using `oldid=1327`)?

File: sensors/environmental-sensors/x86/src/Makefile [hunk 6]
```diff
@@ -2,4 +2,12 @@ ccflags-y += ${PROJECTINCLUDE}
 ccflags-y +=-I$(ZEPHYR_BASE)/include/drivers
 ccflags-y +=-I$(ZEPHYR_BASE)/drivers -I$(ZEPHYR_BASE)/arch/x86
 
+ifneq ($(VERSION_MAJOR), 1)
```
Comment (gvancuts) [inline sensors/environmental-sensors/x86/src/Makefile]: Same comment as above about nesting the `ifneq` statements

File: sensors/environmental-sensors/README.md (hunk 2 above)
Comment (gvancuts) [inline sensors/environmental-sensors/README.md]: Could me move this section towards the beginning of the doc?

File: sensors/environmental-sensors/README.md (hunk 5 above)
Comment (kiendinh) [inline sensors/environmental-sensors/README.md]: We don't use MyNew anymore from 1.6, but move to Zephyr firmware from 1.6

File: sensors/environmental-sensors/README.md (hunk 2 above)
Comment (kiendinh) [inline sensors/environmental-sensors/README.md]: I've just moved it in a re-pushed commit

File: sensors/DC_power_meter/README.md (hunk 4 above)
Comment (kiendinh) [inline sensors/DC_power_meter/README.md]: @gvancuts , I'd prefer to keep the order the same with other READMEs.

File: sensors/environmental-sensors/x86/src/Makefile (hunk 6 above)
Comment (kiendinh) [inline sensors/environmental-sensors/x86/src/Makefile]: @gvancuts , it's been done in the re-pushed commit

**Comment by kiendinh** [comment]
@qiuzhong, the repushed commit has corrected the UV value. Please check. Thanks.

File: sensors/environmental-sensors/README.md (hunk 5 above)
Comment (gvancuts) [inline sensors/environmental-sensors/README.md]: What I meant was that you still have a link to an older revision of the wiki on line 79:
...information from this [wiki](https://wiki.zephyrproject.org/index.php?title=Arduino_101&oldid=975)...

I'm assuming this needs to be updated too?

File: sensors/environmental-sensors/x86/src/Makefile (hunk 6 above)
Comment (gvancuts) [inline sensors/environmental-sensors/x86/src/Makefile]: Thanks!

File: sensors/DC_power_meter/README.md (hunk 4 above)
Comment (gvancuts) [inline sensors/DC_power_meter/README.md]: OK

File: sensors/DC_power_meter/src/Makefile [hunk 7]
```diff
@@ -2,6 +2,12 @@ ccflags-y += ${PROJECTINCLUDE} \
 		-I$(ZEPHYR_BASE)/include/drivers \
 		-I$(ZEPHYR_BASE)/drivers
 
+ifneq ($(VERSION_MAJOR), 1)
+	ifneq ($(VERSION_MINOR), 6)
+		$(error This app works only with Zephyr 1.6)
+	endif
+endif
```
Comment (gvancuts) [inline sensors/DC_power_meter/src/Makefile]: Have you tested this feature?

I've tried very quickly to fake using Zephyr 2.6 (by modifying the top-level Makefile in the zephyr-project folder but that did not trigger any error.

Comment (gvancuts) [inline sensors/DC_power_meter/src/Makefile]: Just ran some more tests... nesting does not seem to work...

Comment (gvancuts) [inline sensors/DC_power_meter/src/Makefile]: Sorry about that @kiendinh , can you revert the changes related to this and go back to your original version. My logic was flawed and the text was not going to be printed twice as `make` would exit anyway at the first occurence of `$(error ...)`

**Comment by qiuzhong** [comment]
@kiendinh , I rebuilt the environmental_sensor code, this time the UV index was 0 in doors. It's sunny today here but the sunshine is not strong. So I also brought the device to outdoor and powered on it under the sun at 13:30 P.M. The UV index still remained 0.  I am not sure if this is the normal behavior.

**Comment by kiendinh** [comment]
@qiuzhong The UV sensor is next to the connector, which is often up-side down. Pls show it towards the sun light. Besides, the window glasses cut most of the UV, so I would recommend to be outdoor completely.

**Comment by qiuzhong** [comment]
@kiendinh , I retried and did show the UV sensor towards the sunshine but nothing changed. I did test the UV sensor outdoors. What UV index value do you usually get?

File: sensors/environmental-sensors/README.md (hunk 5 above)
Comment (kiendinh) [inline sensors/environmental-sensors/README.md]: Oh, I see. I've just re-pushed it. Thanks.

**Comment by kiendinh** [comment]
@qiuzhong , I could get 0 to 5 depending on the conditions.

**Comment by kiendinh** [comment]
@qiuzhong I just rechecked it under the sun and it worked fine. By tilting the angle of the sensor towards the sun, I could get value from 0 to 5. The same results were received and displayed in the other side over BLE.

File: sensors/DC_power_meter/src/Makefile (hunk 1 above)
Comment (kiendinh) [inline sensors/DC_power_meter/src/Makefile]: It's just one message. I've moved it back to the original version.

File: sensors/environmental-sensors/arc/src/Makefile (hunk 3 above)
Comment (kiendinh) [inline sensors/environmental-sensors/arc/src/Makefile]: Same as above.

**Comment by qiuzhong** [comment]
@kiendinh , thanks for the checking. I'll choose a sunny day to retest it again.

**Comment by qiuzhong** [comment]
@kiendinh , I tested the UV sensor this noon as I think it's when the UV rays are the strongest of the day. This time I got a UV value 5!
And other environmental sensors all behave normal as before. The Bluetooth client app can discover the Zephyr Environmental Sensor service too.
So for me, your PR works well.

**Comment by gvancuts** [comment]
Thanks @qiuzhong ! Merging this now
